### PR TITLE
Adopt to logging changes in istio/pkg

### PR DESCRIPTION
**Pull request by howardjohn**
**Please provide a description of this PR:**

### CODE REVIEW

**Review by jacob-delgado — APPROVED** [review]

**Review by therealmitchconnors — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: pilot/pkg/bootstrap/istio_ca.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (s *Server) RunCA(grpc *grpc.Server, ca caserver.CertificateAuthority, opts
 	if err == nil {
 		tok, err := detectAuthEnv(string(token))
 		if err != nil {
-			log.Warn("Starting with invalid K8S JWT token", err, string(token))
+			log.Warnf("Starting with invalid K8S JWT token: %v", err)
```
Comment (costinm) [inline pilot/pkg/bootstrap/istio_ca.go:156]: I really don't like more Warnf and equivalents. 

Much better to add a structured logging method to the log package - for example WarnCtx() to match the new API - and convert the existing Warn() to WarnCtx.

**Comment by costinm** [comment]
I wish this change was properly discussed - with context on otel logging and the new slog upstream API in go.

**Comment by howardjohn** [comment]
Its removing bad usages of the existing API to a smaller surface. When we
have 2 types of logs - Warnf(any...) and Warn(any) - it's far easier to
write a simple query to translate them all to a new API.

Moving 5 logs over to a new API is not useful -- IMO its far worse. There
should be consistency across the codebase.

This moves towards more consistency, which makes it trivial to move the
*entire* codebase to slog later once its released.

**Comment by costinm** [comment]
On Tue, May 2, 2023 at 1:31 PM John Howard ***@***.***> wrote:

> Its removing bad usages of the existing API to a smaller surface. When we
> have 2 types of logs - Warnf(any...) and Warn(any) - it's far easier to
> write a simple query to translate them all to a new API.
>
> Moving 5 logs over to a new API is not useful -- IMO its far worse. There
> should be consistency across the codebase.
>
> This moves towards more consistency, which makes it trivial to move the
> *entire* codebase to slog later once its released
>

Maybe - but it's something that is better discussed a bit.

I don't see how more Warnf() will make anything better - either having a
concatenated
string or printf still makes a non-tool-friendly output. Almost all
upstream logging packages
 allow both styles and I don't know any other package where Printf is
mandatory,
K8s is attempting to make structured logging mandatory - I'm not suggesting
it's bad
to have a consistent style, but if the consistent style is 'arbitrary
strings' using pritnf instead
of concatenating doesn't make any difference.

Either way - not blocking this PR, majority of Warn and Error are printf
style and mostly
useless - this change doesn't make the logs any more or less useful.

**Review by hanxiaop — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Pull request merged**